docs/user-guide/pandas.md

```diff
@@ -0,0 +1,156 @@
+# Pandas Workflow
+
+## Connecting To Tamr
+
+Connecting to a Tamr instance:
+
+```python
+import os
+import pandas as pd
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+username = os.environ['TAMR_USERNAME']
+password = os.environ['TAMR_PASSWORD']
+
+auth = UsernamePasswordAuth(username, password)
+tamr = Client(auth)
+```
+## Load dataset as Dataframe
+
+### Loading: In Memory
+
+Loading a `dataset` as a pandas `dataframe` is possible via the `from_records()` method that pandas provides. 
+An example is shown below:
+
+```python
+my_dataset = tamr.datasets.by_name("my_tamr_dataset")
+df = pd.DataFrame.from_records(my_dataset.records())
+```
+
+This will construct a pandas dataframe based on the records that are streamed in, and stored in the pandas dataframe.
+Once all records have been loaded, you will be able to interact with the dataframe normally. 
+
+Note that as values are typically represented inside `arrays` within Tamr, the values will be encapsulated `lists` 
+inside the dataframe. You can use traditional methods in pandas to deal with this; for example by calling `.explode()`,
+or extracting specific elements. 
+
+### Loading:  Streaming
+When working with large `datasets` it is sometimes better not to work in memory, but to iterate through a dataset. 
+Since `dataset.records()` is a generator, this can easily be done as follows:
+```python
+for record in dataset.records():
+    df = pd.DataFrame.from_records(record)
```

Comment:
This example doesn't work when I test it.  You might need to first create the DataFrame then append rows to it if going this route.  

I don't see the advantage of creating a DataFrame this way unless you're going to use a subset of rows.  In that case, I think you can just pass `nrows` to `DataFrame.from_records(my_dataset.records())`

Or columns, but that is covered below

Reply:
Yeah, the point here is not to fill a data frame, but to access individual records for whatever.

Reply:
Clarified, have a look if it makes sense.

Reply:
@tiems90 something that could help here is showing why wrapping a single record in a dataframe is useful (as opposed to operating directly on the record as a `dict`).

If there is a reason for wrapping in a dataframe, then renaming `df` to `single_record_df` or something would be helpful for legibility.